homeplug_exporter.go

```diff
@@ -101,22 +100,33 @@ func (e *Exporter) collect(ch chan<- prometheus.Metric) error {
 	}
 
 	for _, info := range netinfos {
+		if len(info.Networks) == 0 {
+			continue
+		}
 		for _, network := range info.Networks {
-			ch <- prometheus.MustNewConstMetric(e.network, prometheus.GaugeValue,
-				float64(network.ShortID), hex.EncodeToString(network.NetworkID[:]), strconv.FormatInt(int64(network.TEI), 10), network.CCoAddress.String())
+			ch <- prometheus.MustNewConstMetric(e.network, prometheus.GaugeValue, 1,
+				info.Address.String(),
+				network.NetworkID.String(),
+				strconv.FormatInt(int64(network.ShortID), 10),
+				strconv.FormatInt(int64(network.TEI), 10),
```

Comment:
`strconv.Itoa` is a shorter way of writing `strconv.FormatInt(int64(i), 10)`. The `FormatInt` is really only needed for working with arbitrary / unusual bases, i.e. not base 2, 10 or 16 (which are easily done with `fmt.Sprintf`).
```suggestion
				strconv.Itoa(network.ShortID),
				strconv.Itoa(network.TEI),
```

Reply:
Thanks, I still don't know much go, so I have done a fair bit of copy-pasting here :)

Reply:
Sadly a cast is still needed

Reply:
Right, `strconv.Itoa()` takes an `int` variable. Since that's an _untyped_ `int`, are the values likely to exceed 2^31-1? I've already seen enough 32-bit overflows in the Prometheus ecosystem due to sloppy use of untyped ints tested only on 64-bit archs.

Reply:
You are right. I went the easy way because it was originally like this, but I see now that there is a `strconv.FormatUint` so that should solve this properly.

Reply:
I dug a little deeper and I see that `ShortID`, `TEI` and `CCoTEI` are all `uint8`, so they would not have overflowed an `int` cast when passed to `strconv.Itoa`. But in the grand scheme of things, `strconv.FormatUint` is a more appropriate choice than `strconv.FormatInt` for an operand which is originally unsigned.